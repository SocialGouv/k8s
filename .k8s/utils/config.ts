type ConfigTypes = {
  name: string
  type: string
  subdomain: string
  hasura?: boolean
  azurepg?: boolean
  probesPath?: string
}

const Config = () => {
  const config = require(process.env.SOCIALGOUV_CONFIG_PATH ??
    "../../.socialgouv/config.json")

  return config as ConfigTypes
}

export default Config
