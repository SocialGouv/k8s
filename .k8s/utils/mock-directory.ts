import fs from "fs"
import path from "path"
import fsx from "fs-extra"
import { directory } from "tempy";

export default () => {
  const cwd = directory();

  fs.symlinkSync(path.join(__dirname, "..", "utils"), `${cwd}/utils`)
  fsx.copySync(path.join(__dirname, "..", "components"), `${cwd}/components`)
  fs.symlinkSync(path.join(__dirname, "..", "kosko.toml"), `${cwd}/kosko.toml`)
  fs.symlinkSync(path.join(__dirname, "..", "node_modules"), `${cwd}/node_modules`)
  fs.symlinkSync(path.join(__dirname, "..", "tsconfig.json"), `${cwd}/tsconfig.json`)

  return cwd
}
