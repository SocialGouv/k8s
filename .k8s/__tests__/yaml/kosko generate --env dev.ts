//

import fsx from "fs-extra";
import path from "path";
import { getEnvManifests } from "@socialgouv/kosko-charts/testing";
import { project } from "@socialgouv/kosko-charts/testing/fake/gitlab-ci.env";

// changes the cwd to a tmp folder
import "../../utils/mock-directory";

jest.setTimeout(1000 * 60);

test("yaml: kosko generate --dev", async () => {
  const name = "myapp";
  const destFolder = `${process.cwd()}/environments/dev/yaml`;
  const sourceFile = path.join(__dirname, "redirect.yml");

  await fsx.mkdir(destFolder, { recursive: true });
  await fsx.copyFile(sourceFile, `${destFolder}/redirect.yml`);

  expect(
    await getEnvManifests("dev", "'!(_*)'", {
      ...project(name).dev,
      SOCIALGOUV_CONFIG_PATH: __dirname + "/config.json",
    })
  ).toMatchSnapshot();
});
