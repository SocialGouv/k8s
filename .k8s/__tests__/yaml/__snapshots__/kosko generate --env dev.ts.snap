// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`yaml: kosko generate --dev 1`] = `
"---
apiVersion: apps/v1
kind: Deployment
metadata:
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  namespace: myapp-85-master-dev2
spec:
  replicas: 1
  selector:
    matchLabels:
      app: myapp
  template:
    metadata:
      annotations:
        kapp.k14s.io/disable-default-ownership-label-rules: ''
        kapp.k14s.io/disable-default-label-scoping-rules: ''
        app.gitlab.com/app: socialgouv-myapp
        app.gitlab.com/env: master-dev2
        app.gitlab.com/env.name: master-dev2
      labels:
        app: myapp
        application: master-dev2-myapp
        owner: myapp
        team: myapp
        cert: wildcard
    spec:
      containers:
        - image: >-
            harbor.fabrique.social.gouv.fr/myapp/myapp:0123456789abcdefghijklmnopqrstuvwxyz0123
          livenessProbe:
            failureThreshold: 6
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 30
            periodSeconds: 5
            timeoutSeconds: 5
          name: myapp
          ports:
            - containerPort: 3000
              name: http
          readinessProbe:
            failureThreshold: 15
            httpGet:
              path: /healthz
              port: http
            initialDelaySeconds: 0
            periodSeconds: 5
            successThreshold: 1
            timeoutSeconds: 1
          resources:
            limits:
              cpu: 200m
              memory: 256Mi
            requests:
              cpu: 50m
              memory: 128Mi
          startupProbe:
            failureThreshold: 12
            httpGet:
              path: /healthz
              port: http
            periodSeconds: 5
          env:
            - name: APP_BASE_URL
              value: https://master-dev2-myapp.dev2.fabrique.social.gouv.fr
---
apiVersion: v1
kind: Service
metadata:
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  annotations:
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  namespace: myapp-85-master-dev2
spec:
  ports:
    - name: http
      port: 80
      targetPort: 3000
  selector:
    app: myapp
  type: ClusterIP
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    kapp.k14s.io/disable-default-ownership-label-rules: ''
    kapp.k14s.io/disable-default-label-scoping-rules: ''
    app.gitlab.com/app: socialgouv-myapp
    app.gitlab.com/env: master-dev2
    app.gitlab.com/env.name: master-dev2
  labels:
    app: myapp
    application: master-dev2-myapp
    owner: myapp
    team: myapp
    cert: wildcard
  name: myapp
  namespace: myapp-85-master-dev2
spec:
  rules:
    - host: master-dev2-myapp.dev2.fabrique.social.gouv.fr
      http:
        paths:
          - backend:
              serviceName: myapp
              servicePort: 80
            path: /
  tls:
    - hosts:
        - master-dev2-myapp.dev2.fabrique.social.gouv.fr
      secretName: wildcard-crt
---
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  annotations:
    kubernetes.io/ingress.class: nginx
    nginx.ingress.kubernetes.io/permanent-redirect: https://testapp.fabrique.social.gouv.fr$request_uri
    certmanager.k8s.io/cluster-issuer: letsencrypt-prod
    kubernetes.io/tls-acme: 'true'
  labels:
    app.kubernetes.io/component: redirect
    app.kubernetes.io/name: www
  name: www-redirect
  namespace: myapp-85-master-dev2
spec:
  rules:
    - host: www.testapp.fr
    - host: testapp.fr
  tls:
    - hosts:
        - www.testapp.fr
        - testapp.fr
      secretName: www-redirect
"
`;
